timesketch/tsctl.py

```diff
@@ -616,3 +622,91 @@ def validate_context_links_conf(path):
             print(f'=> OK: "{entry}"')
         except (ValidationError, SchemaError) as err:
             print(f'=> ERROR: "{entry}" >> {err}\n')
+
+
+# Analyzer stats cli command
+@cli.command(name="analyzer-stats")
+@click.argument("analyzer_name")
+@click.option(
+    "--timeline_id",
+    required=False,
+    help="Timeline ID if the analyzer results should be filtered by timeline.",
+)
+@click.option(
+    "--scope",
+    required=False,
+    help="Scope on: [many_hits, long_runtime, recent]",
+)
+@click.option(
+    "--result_text_search",
+    required=False,
+    help="Search in result text. E.g. for a specific rule_id.",
+)
+@click.option(
+    "--limit",
+    required=False,
+    help="Limit the number of results.",
+)
+def sigma_stats(analyzer_name, timeline_id, scope, result_text_search, limit):
+    """Prints sigma analyzer stats."""
+
+    if timeline_id:
+        timeline = Timeline.query.get(timeline_id)
+        if not timeline:
+            print("No timeline found with this ID.")
+            return
+        analysis_history = Analysis.query.filter_by(
+            timeline=timeline, analyzer_name=analyzer_name
+        ).all()
+    elif analyzer_name == "all":
+        analysis_history = Analysis.query.filter_by().all()
+    else:
+        # analysis filter by analyzer_name is sigma
+        analysis_history = Analysis.query.filter_by(analyzer_name=analyzer_name).all()
+
+    df = pd.DataFrame()
+    for analysis in analysis_history:
+        if analysis.analyzer_name in [analyzer_name, "all"]:
+            # extract number of hits from result to a int so it could be sorted
+            try:
+                matches = re.search(r"\d+(?=\s+events)", analysis.result)
+            except TypeError:
+                matches = 0
+            numbers = 0
+            if matches:
+                numbers = int(matches.group())
```

Comment:
You could do the int conversion already in line 672 above and then have matches as resulting integer only. e.g.
```
try:
  matches = int(re.search(r"\d+(?=\s+events)", analysis.result).group())
except TypeError:
  matches = 0
```